js/main.js

```diff
@@ -69,18 +67,42 @@ var $newEntry = document.querySelector('div[data-view="entry-form"]');
 var $entriesLink = document.querySelector('a[href="#entry-form"]');
 
 $newButton.addEventListener('click', function () {
-  $newEntry.className = '';
-  $entriesPage.className = 'hidden';
-  data.view = "entry-form"
+  switchViews($newEntry.getAttribute('data-view'));
+  data.view = 'entry-form';
 });
 
 $entriesLink.addEventListener('click', function () {
-  $newEntry.className = 'hidden';
-  $entriesPage.className = '';
-  data.view = "all-entries"
+  switchViews($entriesPage.getAttribute('data-view'));
+  data.view = 'all-entries';
 });
 
+var $viewElements = document.querySelectorAll('div[data-view]');
+
+function switchViews(view) {
+  for (var i = 0; i < $viewElements.length; i++) {
+    if ($viewElements[i].getAttribute('data-view') !== view) {
+      $viewElements[i].className = 'hidden';
+    } else {
+      $viewElements[i].className = '';
+    }
+  }
+}
+
+window.addEventListener('DOMContentLoaded', appendDOM);
 var $noItems = document.querySelector('.center-text');
 if ($ul.childElementCount === 0) {
   $noItems.className = 'center-text hidden';
+} else {
+  $noItems.className = 'center-text';
+}
+
+function appendDOM() {
+  for (var i = 0; i < data.entries.length; i++) {
+    entryDOM(data.entries[i]);
+  }
+  if (data.view === 'entry-form') {
+    switchViews($newEntry.getAttribute('data-view'));
+  } else {
+    switchViews($entriesPage.getAttribute('data-view'));
+  }
```

Comment:
You should just call `switchViews` using the current value of `data.view` as the argument, no conditional necessary 🤓